server/app/mutations/grid_services/update.rb

```diff
@@ -120,7 +120,7 @@ def build_grid_service_envs(env)
       new_env = GridService.new(env: env).env_hash
       current_env = self.grid_service.env_hash
       new_env.each do |k, v|
-        if v.empty? && current_env[k]
+        if (v.nil? || v.empty?) && current_env[k]
```

Comment:
I was thinking about having invalid values there because of earlier bugs, this should avoid crashing in case you installed a stack with invalid envs.